### ISSUE TITLE
ERR_OSSL_DSO_COULD_NOT_LOAD_THE_SHARED_LIBRARY

### ISSUE THREAD

**Issue by FelippeChemello** [theoephraim/node-google-spreadsheet#553]
I'm getting this error 
```js
node:internal/crypto/sig:131
  const ret = this[kHandle].sign(data, format, type, passphrase, rsaPadding,

Error: error:25066067:DSO support routines:dlfcn_load:could not load the shared library
    at Sign.sign
    at Object.sign
    at Object.jwsSign [as sign]
    at GoogleToken.requestToken
    at GoogleToken.getTokenAsyncInner
    at GoogleToken.getTokenAsync
    at GoogleToken.getToken
    at JWT.refreshTokenNoCache
    at JWT.refreshToken
    at JWT.authorizeAsync
  opensslErrorStack: [
    'error:0E076071:configuration file routines:module_run:unknown module name',
    'error:0E07506E:configuration file routines:module_load_dso:error loading dso',
    'error:25070067:DSO support routines:DSO_load:could not load the shared library'
  ],
  library: 'DSO support routines',
  function: 'dlfcn_load',
  reason: 'could not load the shared library',
  code: 'ERR_OSSL_DSO_COULD_NOT_LOAD_THE_SHARED_LIBRARY'
}
```
Code:
```js
const sheets = new GoogleSpreadsheet(SPREADSHEET_ID);

    await sheets.useServiceAccountAuth({
        client_email: GOOGLE_EMAIL,
        private_key: GOOGLE_PRIVATE_KEY,
    });
```

Already tried to install OPENSSL 1.1.1 as reccomended on [this issue](https://github.com/theoephraim/node-google-spreadsheet/issues/537), however didn't work.

```shell
$ openssl version
OpenSSL 1.1.1c  28 May 2019
```

**Comment by theoephraim** [comment]
hrmmm thats a new one to me... What system are you running this on?

**Comment by williamluke4** [comment]
Just ran into the same issue, 
```
OS: Ubuntu 22.04 LTS x86_64 
Kernel: 5.15.0-33-generic
Node: v16.13.2
OpenSSL:  3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)
google-spreadsheet: 3.3.0
```
The issue occurs locally and when build a nextjs project on vercel servers

**Comment by williamluke4** [comment]
Setting `OPENSSL_CONF` variable to `/dev/null` works but :grimacing:

**Comment by soundproofboot** [comment]
I also got this error using Ubuntu 22.04 with WSL2 on WIndows 10, also in a next.js project. I don't get the error using Ubuntu20.04 if that helps.

**Comment by DanielBailey-web** [comment]
I just got this error too. I am running:
```
Operating System:
      Platform: linux
      Arch: x64
      Version: #34-Ubuntu SMP Wed May 18 13:34:26 UTC 2022
    Binaries:
      Node: 14.18.1
      npm: 8.6.0
      Yarn: 1.22.18
      pnpm: N/A
    Relevant packages:
      next: 12.1.6
      react: 18.1.0
      react-dom: 18.1.0
      google-spreadsheet: 3.3.0
```

**Comment by williamluke4** [comment]
@theoephraim Any what's going on here?

**Comment by DanielBailey-web** [comment]
I am getting the error still, but it is no longer fatal after my latest ubuntu update. it just logs it and seems to not care past that. Also, it seems to be operating find in my production environment ([Docker] FROM node:16-alpine AS deps. Running on Cloud Run) So this is definitely a weird one.

**Comment by theoephraim** [comment]
no idea whats going on sorry and I sort of doubt it is caused _directly_ by this package.
Seems like it may have something to do with the 'googleapis' module, but of course they have issues disabled on github... 

Maybe some useful info here: https://stackoverflow.com/questions/70102543/error-could-not-load-the-shared-library-api-google-vms

**Comment by nicolas-t** [comment]
Fixed this issue on Ubuntu 22.04 by using **node 18**

**Comment by theoephraim** [comment]
closing for now. Please reopen if still running into issues with the latest version.
